src/main/java/io/cdap/plugin/gcp/common/GCPUtils.java

```diff
@@ -163,6 +169,10 @@ public static String extractPrivateKey(ServiceAccountCredentials credentials) {
   public static BigQuery getBigQuery(String project, @Nullable Credentials credentials) {
     BigQueryOptions.Builder bigqueryBuilder = BigQueryOptions.newBuilder().setProjectId(project);
     if (credentials != null) {
+      if (credentials instanceof ServiceAccountCredentials
+            && ((ServiceAccountCredentials) credentials).getScopes().isEmpty()) {
```

Comment:
is that possible that the scope is not empty but doesn't contain the drive scope ?

Reply:
Yea, I modified the logic to always add these scopes